cpp/tests/CMakeLists.txt

```diff
@@ -96,9 +96,16 @@ function(ConfigureTest CMAKE_TEST_NAME CMAKE_TEST_SRC CMAKE_EXTRA_LIBS)
 ###       BUILD_RPATH "${TARGET_BUILD_RPATH}")
         ${OpenMP_CXX_LIB_NAMES})
     endif(OpenMP_CXX_FOUND)
-    set_target_properties(${CMAKE_TEST_NAME} PROPERTIES
-        RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/gtests/")
 
+    # CUDA_ARCHITECTURES=OFF implies cmake will not pass arch flags to the
+    # compiler. CUDA_ARCHITECTURES must be set to a non-empty value to prevent
+    # cmake warnings about policy CMP0104. With this setting, arch flags must be
+    # manually set! ("evaluate_gpu_archs(GPU_ARCHS)" is the current mechanism
+    # used in cpp/CMakeLists.txt for setting arch options).
+    # Run "cmake --help-policy CMP0104" for policy details.
+    set_target_properties(${CMAKE_TEST_NAME} PROPERTIES
```

Comment:
This will work for this single target. I would have thought that other targets that use CUDA would generate `CMP0104` warnings as well. 

My initial thought is that `cpp/CMakeLists.txt` around line 95 should set `CMAKE_CUDA_ARCHITECTURES` to OFF. When we require CMake 3.18 we review using `CMAKE_CUDA_ARCHITECTURES` or continue disabling it.

Reply:
Thanks, Robert. It's not obvious from the diff, but that line is part of a function which is applied for every test target which takes care of the `CMP0104` warnings for all test binaries.